mitodl/ocw-studio
# Mass publish sites management command

#### Pre-Flight checklist

#### What are the relevant tickets?
Closes #730 
Closes #805

#### What's this PR do?
- Renames the `trigger_pipelines` command to `mass_publish` and refactors it to run batched sets of tasks to:
  - run prepublish actions (but only if specifically requested)
  - sync content to github
  - merge git content to the appropriate branch
  - trigger the concourse pipeline
  - filters to `source=ocw-import` by default unless overridden
- Renames some functions to hopefully make the code more consistent and less confusing  
- Does not try to update youtube metadata for videos imported from legacy OCW (unless on production).

#### How should this be manually tested?
- Use the same `AWS_`, `CONCOURSE_`,  but not`GIT` settings as RC (use a personal org)
- Import some ocw sites: `manage.py import_ocw_course_sites --bucket ocw-to-hugo-output-qa --limit 2`
- Run `manage.py mass_publish live --filter 1-1 --chunks 1  --prepublish`
- The git 'preview' and 'release' branches of matching sites should be merged from main, and the concourse pipelines should be triggered.  They will fail if pointing to your local org repo but that's ok.  
- Once testing is done, from a heroku shell run  `manage.py import_ocw_course_sites --bucket ocw-to-hugo-output-qa --limit 2` again to set back to the correct org repo.
